src/differentiation/compute_jacobian_ad.jl

```diff
@@ -44,20 +44,34 @@ end
 generate_chunked_partials(x,color,N::Integer) = generate_chunked_partials(x,color,Val(N))
 function generate_chunked_partials(x,color,::Val{N}) where N
 
-    # TODO: should only go up to the chunksize each time, and should
-    # generate p[i] different parts, each with less than the chunksize
+    chunksize = getsize(default_chunk_size(maximum(color)))
```

Comment:
we still aren't making use of the actual chunksize if passed by the user, so we should fix that up in a follow up.